Package request: python3-apt

<!-- Important note: motivating 'why this package is needed' will most likely give the package higher priority in the developers TODO-list -->

**Package description**
This package provides APT python bindings required for Ansible. Without this, the `apt:` module cannot be used.

**Link to home page and sources**
1. Home page: https://packages.debian.org/search?keywords=python-apt
2. Source code: https://salsa.debian.org/apt-team/python-apt

**Additional information**
Have you compiled or tried to compile the package on device?

Yes. Without any patches the build fails with:
```
[...]
In file included from python/acquire.cc:26:
python/apt_pkgmodule.h:14:10: fatal error: 'apt-pkg/hashes.h' file not found
#include <apt-pkg/hashes.h>
         ^~~~~~~~~~~~~~~~~~
1 error generated.
error: command 'aarch64-linux-android-clang' failed with exit status 1
```

NOTE: The headers are provided by `libapt-pkg-dev` (not available on Termux either)

--- a/packages/python-apt/build.sh
+++ b/packages/python-apt/build.sh
@@ -1,0 +1,33 @@
+TERMUX_PKG_HOMEPAGE=https://apt-team.pages.debian.net/python-apt/
+TERMUX_PKG_DESCRIPTION="Python bindings for APT"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=2.3.0
+TERMUX_PKG_SRCURL=https://ftp.debian.org/debian/pool/main/p/python-apt/python-apt_${TERMUX_PKG_VERSION}.tar.xz
+TERMUX_PKG_SHA256=de4a284bc7a738793615631d451524f42404034f4d787953b7621b25bc29e474
+TERMUX_PKG_DEPENDS="apt, libc++, python"
+TERMUX_PKG_BUILD_IN_SRC=true
+
+_PYTHON_VERSION=$(. $TERMUX_SCRIPTDIR/packages/python/build.sh; echo $_MAJOR_VERSION)
+
+termux_step_pre_configure() {
+	termux_setup_python_crossenv
+	pushd $TERMUX_PYTHON_CROSSENV_SRCDIR
+	_CROSSENV_PREFIX=$TERMUX_PKG_BUILDDIR/python-crossenv-prefix
+	python${_PYTHON_VERSION} -m crossenv \
+		$TERMUX_PREFIX/bin/python${_PYTHON_VERSION} \
+		${_CROSSENV_PREFIX}
+	popd
+	. ${_CROSSENV_PREFIX}/bin/activate
+
+	pushd ${_CROSSENV_PREFIX}/build/lib/python${_PYTHON_VERSION}/site-packages
+	patch --silent -p1 < $TERMUX_PKG_BUILDER_DIR/setuptools-44.1.1-no-bdist_wininst.diff || :
+	popd
+
+	LDFLAGS+=" -lpython${_PYTHON_VERSION}"
+}
+
+termux_step_make_install() {
+	DEBVER=$TERMUX_PKG_VERSION \
+		python setup.py install --force --prefix $TERMUX_PREFIX
+}

--- a/packages/python-apt/setuptools-44.1.1-no-bdist_wininst.diff
+++ b/packages/python-apt/setuptools-44.1.1-no-bdist_wininst.diff
@@ -1,0 +1,35 @@
+Copyright (C) 2016 Jason R Coombs <jaraco@jaraco.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the "Software"), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+of the Software, and to permit persons to whom the Software is furnished to do
+so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+
+--- a/setuptools/command/install_scripts.py
++++ b/setuptools/command/install_scripts.py
+@@ -32,12 +32,7 @@
+         )
+         bs_cmd = self.get_finalized_command('build_scripts')
+         exec_param = getattr(bs_cmd, 'executable', None)
+-        bw_cmd = self.get_finalized_command("bdist_wininst")
+-        is_wininst = getattr(bw_cmd, '_is_running', False)
+         writer = ei.ScriptWriter
+-        if is_wininst:
+-            exec_param = "python.exe"
+-            writer = ei.WindowsScriptWriter
+         if exec_param == sys.executable:
+             # In case the path to the Python executable contains a space, wrap
+             # it so it's not split up.

